lib/shared/inference/openai_image_generator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ::DiscourseAi
+  module Inference
+    class OpenAiImageGenerator
+      TIMEOUT = 60
+
+      def self.perform!(prompt, model: "dall-e-3", size: "1024x1024", api_key: nil)
+        api_key ||= SiteSetting.ai_openai_api_key
+
+        url = URI("https://api.openai.com/v1/images/generations")
+        headers = { "Content-Type" => "application/json" }
+        headers["Authorization"] = "Bearer #{api_key}"
+
+        payload = { model: model, prompt: prompt, n: 1, size: size, response_format: "b64_json" }
```

Comment:
What is `n: 1` here?

Reply:
number of images... but it is meant to only be 1 for now ... I prefer not to omit in case they pick a diff default.